src/sorting/SortingQuestions.java

```diff
@@ -25,6 +25,34 @@
         return arr;
     }
 
+    //2. Implement selection sort O(n^2)
+    /*
+    This sorting algorithm is an in-place comparison-based algorithm in which the list is divided into two parts,
+    the sorted part at the left end and the unsorted part at the right end. Initially, the sorted part is empty
+    and the unsorted part is the entire list.
+     */
+    public static int[] selectionSort(int[] arr){
+        int length = arr.length;    //size of the array
+        int indexOfMinElement;
+        int temp;
+
+        for (int i = 0; i < length; i++){
+            indexOfMinElement = i;  //assuming the ith element to be the smallest at start of the inner loop
+            for (int j = i + 1; j < length; j++){
+                if (arr[j] < arr[indexOfMinElement]){
+                    //updating with the next minimum value index
+                    indexOfMinElement = j;
+                }
+
+                //swapping the current minimum to be in the sorted portion of the array
+                temp = arr[i];
```

Comment:
In place swap must happen after the second loop. Insertion sort improves over bubble sort in the fact that the number of swaps is reduced. As of right now, you are swapping the elements after every comparison therefore it is still bubble sort.

Reply:
@neoprez thank you! I see the difference between bubble and selection